h2o-hadoop/assemblyjar.gradle

```diff
@@ -13,10 +13,10 @@ dependencies {
   else {
     compile('org.apache.hadoop:hadoop-client:' + hadoopMavenArtifactVersion)
   }
-  compile project(':h2o-hadoop:h2o-' + hadoopVersion)
-  compile project(':h2o-app')
-  compile project(":h2o-web")
-  compile project(":h2o-avro-parser")
+  compile project(':h2o-hadoop:h2o-' + hadoopVersion) // this is a magically hidden dependency on h2odriver.jar
```

Comment:
I don't understand - what is the magic going on here? The assembly is responsible for making the h2odriver.jar

Reply:
I think your understanding is correct

> However, discovering that SECOND depends on FIRST requires decrypting the build code, instead of having the dependency clearly stated in the SECOND module's build file.

Yep, the dependency is stated in the "parent" generic file, not in the specific hadoop assembly build script.

Simplifying the build is very welcome. We want the build to be easily understandable to anyone.